scarlet/source.py

```diff
@@ -72,159 +213,168 @@ def __init__(self, model_frame, sky_coord, observations):
-        starlet_thresh=5,
         min_grad=0.1,
+        starlet_thresh=5,
     ):
         """Extended source intialized to match a set of observations
 
         Parameters
         ----------
-        frame: `~scarlet.Frame`
+        model_frame: `~scarlet.Frame`
             The frame of the model
         sky_coord: tuple
             Center of the source
         observations: instance or list of `~scarlet.observation.Observation`
             Observation(s) to initialize this source.
-        obs_idx: int
-            Index of the observation in `observations` to
-            initialize the morphology.
         thresh: `float`
             Multiple of the backround RMS used as a
             flux cutoff for morphology initialization.
-        shifting: `bool`
-            Whether or not a subpixel shift is added as optimization parameter
         """
-        center = np.array(frame.get_pixel(sky_coord), dtype="float")
-        self.pixel_center = tuple(np.round(center).astype("int"))
-
-        # initialize SED from sky_coord
-        try:
-            iter(observations)
-        except TypeError:
-            observations = [observations]
 
         # initialize from observation
-        sed, image_morph, bbox = init_extended_source(
+        sed, morph, bbox, thresh = init_starlet_source(
             sky_coord,
-            frame,
+            model_frame,
             observations,
-            coadd=coadd,
-            bg_cutoff=bg_cutoff,
+            coadd,
+            bg_cutoff,
             thresh=thresh,
             symmetric=True,
-            monotonic=True,
+            monotonic="flat",
             min_grad=min_grad,
+            starlet_thresh=starlet_thresh,
         )
-        noise =[]
-        for obs in observations:
-            noise += [mad_wavelet(obs.images) * \
-                    np.sqrt(np.sum(obs._diff_kernels.image**2, axis = (-2,-1)))]
-        noise = np.concatenate(noise)
-        # Threshold in units of noise
-        thresh = starlet_thresh * np.sqrt(np.sum((sed*noise) ** 2))
+        spectrum = TabulatedSpectrum(sed)
+        morphology = StarletMorphology(morph, bbox[1:], thresh)
+        super().__init__(model_frame, spectrum, morphology)
 
-        # Starlet transform of morphologies (n1,n2) with 4 dimensions: (1,lvl,n1,n2), lvl = wavelet scales
-        self.transform = Starlet(image_morph)
-        #The starlet transform is the model
-        morph = self.transform.coefficients
-        # wavelet-scale norm
-        starlet_norm = self.transform.norm
-        #One threshold per wavelet scale: thresh*norm
-        thresh_array = np.zeros(morph.shape) + thresh
-        thresh_array = thresh_array * np.array([starlet_norm])[..., np.newaxis, np.newaxis]
-        # We don't threshold the last scale
-        thresh_array[:,-1,:,:] = 0
+        # since we use the starlet for localized source: it has a center
+        self._center = np.array(model_frame.get_pixel(sky_coord), dtype="float")
```

Comment:
Even when we use a localised model, the wavelet model has a box that give the information about localisation, but the center is somewhat irrelevant.

Reply:
Correct. I'm OK with removing it.

Reply:
I think the notion of a center for starlets is slightly misleading. The point of the starlet is to have something NOT localised in its box, so although I won't fight if you want to keep it, I'd rather do away with it.

Reply:
removed `center`